test/governance/BaseWeightedMultisig.js

```diff
@@ -0,0 +1,362 @@
+
+            expect(validPreviousSigners.length).to.be.equal(OLD_KEY_RETENTION);
+
+            await Promise.all(
+                validPreviousSigners.map(async (signers, index) => {
+                    const isCurrentSigners = await newMultisig.validateProof(
+                        message,
+                        await getWeightedSignaturesProof(
+                            data,
+                            signers,
+                            signers.map(() => index + 2),
+                            (index + 2) * 2,
+                            signers,
+                        ),
+                    );
+                    expect(isCurrentSigners).to.be.equal(false);
+                }),
+            );
+        });
+
+        it('reject the proof from the signers older than key retention', async () => {
+            const data = '0x123abc123abc';
+            const message = hashMessage(arrayify(keccak256(data)));
+            const invalidPreviousSigners = previousSigners.slice(0, -OLD_KEY_RETENTION);
+
+            await Promise.all(
+                invalidPreviousSigners.map(async (signers) => {
+                    await expectRevert(
+                        async (gasOptions) =>
+                            newMultisig.validateProof(
+                                message,
+                                await getWeightedSignaturesProof(
+                                    data,
+                                    signers,
+                                    signers.map(() => 1),
+                                    threshold,
+                                    signers.slice(0, threshold),
+                                ),
+                                gasOptions,
+                            ),
+                        multisig,
+                        'InvalidSigners',
+                    );
+                }),
+            );
+        });
+    });
+
+    describe('transferSignership', () => {
+        it('should allow owner to transfer signership', async () => {
+            const newSigners = [
+                '0x6D4017D4b1DCd36e6EA88b7900e8eC64A1D1315b',
+                '0xb7900E8Ec64A1D1315B6D4017d4b1dcd36E6Ea88',
+            ];
+
+            await expect(multisig.rotateSigners([newSigners, newSigners.map(() => 1), 2])).to.emit(
+                multisig,
+                'SignersRotated',
+            );
+        });
+
+        it('should revert if new signers length is zero', async () => {
+            const newSigners = [];
+
+            await expectRevert(
+                (gasOptions) => multisig.rotateSigners([newSigners, newSigners.map(() => 1), 2], gasOptions),
+                multisig,
+                'InvalidSigners',
+            );
+        });
+
+        it('should not allow transferring signership to address zero', async () => {
+            const newSigners = [ethers.constants.AddressZero, '0x6D4017D4b1DCd36e6EA88b7900e8eC64A1D1315b'];
```

Comment:
```suggestion
            const newSigners = [AddressZero, '0x6D4017D4b1DCd36e6EA88b7900e8eC64A1D1315b'];
```
Can import AddressZero under constants similar to ethers utils